crates/flox/src/commands/channel.rs

```diff
@@ -373,17 +379,73 @@ fn construct_show_params(search_term: &str, flox: &Flox) -> Result<SearchParams>
     })
 }
 
-fn render_show(search_results: &[SearchResult]) -> Result<()> {
-    // FIXME: Proper rendering is coming later
+fn render_show(search_results: &[SearchResult], all: bool) -> Result<()> {
+    let mut pkg_name = None;
+    let mut results = Vec::new();
+    // Collect all versions of the top search result
     for package in search_results.iter() {
-        let pkg_name = package.pkg_subpath.join(".");
-        let description = package
-            .description
-            .as_ref()
-            .map(|d| d.replace('\n', " "))
-            .unwrap_or("<no description provided>".into());
-        println!("{} - {}", pkg_name, description);
+        let this_pkg_name = package.pkg_subpath.join(".");
+        if pkg_name.is_none() {
+            pkg_name = Some(this_pkg_name.clone());
+        }
+        if pkg_name == Some(this_pkg_name) {
+            results.push(package);
+        }
+        // let description = package
+        //     .description
+        //     .as_ref()
+        //     .map(|d| d.replace('\n', " "))
+        //     .unwrap_or("<no description provided>".into());
+        // println!("{} - {}", this_pkg_name, description);
     }
+    if results.is_empty() {
+        // This should never happen since we've already checked that the
+        // set of results is non-empty.
+        bail!("no packages found");
+    }
+    let pkg_name = pkg_name.unwrap();
+    let description = results[0]
+        .description
+        .as_ref()
+        .map(|d| d.replace('\n', " "))
+        .unwrap_or(DEFAULT_DESCRIPTION.into());
+    let versions = if all {
+        let multiple_versions = results
+            .iter()
+            .filter_map(|sr| {
+                // Don't show a "latest" search result, it's just
+                // a duplicate
+                if sr.subtree == Subtree::Catalog
+                    && sr
+                        .abs_path
+                        .last()
+                        .map(|version| version == "latest")
+                        .unwrap_or(false)
+                {
+                    return None;
+                }
+                let name = sr.pkg_subpath.join(".");
+                // TODO: what if one of the search results doesn't have a version?
```

Comment:
Be sure that we actually fix this todo.

It's cool if you want to file a new issue to do it later, but I imagine it's less effort to just write the loop now?

Reply:
It's more of a "I don't know what we should do" if this happens. If I get 4 results for `hello` and one of them doesn't have a `version' field should the output look like this (note the last `hello`)?
```
hello - A program that produces a familiar, friendly greeting
    hello - hello@2.12.1, hello@2.12, hello
```

Reply:
Actually if any have a version then omit unversioned results.

The resolver will never rank them above the versioned ones so they are effectively unreachable for GA.